net-utils/src/ip_echo_client.rs

```diff
@@ -208,12 +231,23 @@ pub(crate) async fn verify_all_reachable_udp(
         warn!("No ports provided for verify_all_reachable_udp to check");
         return true;
     }
+    // Extract the bind_address for requests from the first socket, it should be same for all others too
+    let bind_address = sockets[0]
+        .local_addr()
+        .expect("Sockets should be bound")
+        .ip();
     // This function may get fed multiple sockets bound to the same port.
     // In such case we need to know which sockets are bound to each port,
     // as only one of them will receive a packet from echo server
     let mut ports_to_socks_map: BTreeMap<_, _> = BTreeMap::new();
     sockets.iter().for_each(|&socket| {
-        let port = socket.local_addr().expect("Sockets should be bound").port();
+        let local_binding = socket.local_addr().expect("Sockets should be bound");
```

Comment:
you do similar checks in the function above. Maybe move to a new function? Also in one case `for_each` is used in the other `for`.